toolkit/tools/pkg/rpmssnapshot/rpmssnapshot.go

```diff
@@ -53,42 +52,38 @@ const (
 	rpmSpecBuiltRPMRegexDistributionIndex
 	rpmSpecBuiltRPMRegexArchitectureIndex
 	rpmSpecBuiltRPMRegexMatchesCount
+
+	chrootName = "rpmssnapshot_chroot"
+	runChecks  = false
 )
 
 type SnapshotGenerator struct {
-	chroot        *safechroot.Chroot
-	buildDirPath  string
-	workerTarPath string
+	simplechroottool.SimpleChrootTool
 }
 
-// New creates an unitialized RPMs snapshot generator.
-func New(buildDirPath, workerTarPath string) *SnapshotGenerator {
-	return &SnapshotGenerator{
-		buildDirPath:  buildDirPath,
-		workerTarPath: workerTarPath,
-	}
+// New creates a new snapshot generator.
+func New(buildDirPath, workerTarPath, specsDirPath, distTag string) (newSnapshotGenerator *SnapshotGenerator, err error) {
```

Comment:
> specsDirPath, distTag string [](http://example.com/codeflow?start=37&length=29)

Comment related to the one about the init-clean-up one: if we init here and already pass the spec path and dist tag, then `GenerateSnapshot` shouldn't need these arguments and we can remove them from that method. And, similarly, we can also move everything back to `GenerateSnapshot`. Either way sounds good to me.

Reply:
Good callout, these were only being used for logging so I just moved the log calls outside the function.